ui/src/app/job-list/table/table.component.ts

```diff
@@ -200,4 +200,26 @@ export class JobsTableComponent implements OnInit {
       this.jobs.forEach(j => this.selection.select(j));
     }
   }
+
+  updateCheckBoxSelection(clickedJob: QueryJobsResult, event: MouseEvent): void {
+    // if the user has shift-clicked on a job, find the last selected job and
+    // select all jobs between the two
+    if (event.shiftKey && this.selection.selected.length) {
+      const lastJobClicked = this.selection.selected[this.selection.selected.length - 1];
+      this.selection.select(clickedJob);
+      let indexes = [this.jobs.findIndex(j => j.id === lastJobClicked.id),
+        this.jobs.findIndex(j => j.id === clickedJob.id)];
+      indexes.sort();
+      var startSelecting = false;
+      for (let job of this.jobs) {
```

Comment:
I think it would be clearer to use index iteration here, then drop all of the ID comparisons and job ID matching inside the loop:

```
for (let i = indexes[0] + 1; i < indexes[1]-1; i++) {
  this.selection.select(this.jobs[i]);
}
```

Reply:
good point – done